qemu/tests/qemu_guest_agent_snapshot.py

```diff
@@ -48,8 +49,11 @@ def _action_after_fsfreeze(self, *args):
             image_params = self.params.object_params(image_tag)
             snapshot_test = LiveSnapshot(self.test, self.params,
                                          self.env, image_tag)
+            snapshot_create = BlockDevSnapshotTest(self.test, self.params,
+                                         self.env)
             error_context.context("Creating snapshot", logging.info)
-            snapshot_test.create_snapshot()
+            snapshot_create.prepare_snapshot_file()
+            snapshot_create.create_snapshot()
             error_context.context("Checking snapshot created successfully",
                                   logging.info)
             snapshot_test.check_snapshot()
```

Comment:
Mix `LiveSnapshot` and `BlockDevSnapshotTest` in the same case is not a good idea.
The current way is: `snapshot_create.prepare_snapshot_file()` --> `snapshot_create.create_snapshot()` --> `snapshot_test.check_snapshot()`, it would look strange.

Reply:
What about defining check_snapshot as a member function of QemuGuestAgentSnapshotTest, then we can remove snapshot_test, i.e. LiveSnapshot can be removed, just FYI.

Reply:
updated to only use BlockDevSnapshotTest